### DISCUSSION TITLE
vlib.time: How to implement `time.new_ticker(time.Duration, fn ())` or `set_timer(time.Duration, fn ())`?

### DISCUSSION THREAD

**Discussion by changrui** [vlang/v#12708] [Ideas]
as title, just like `javascript or MFC`, https://www.w3schools.com/jsref/met_win_setinterval.asp
https://docs.microsoft.com/en-us/windows/win32/winmsg/using-timers

**Comment by changrui** [comment]
Test Code:
```
import time

fn ticker() {
	println('logging...')
}

fn set_tick(ticks int, func fn ()) {
	for {
        if time.ticks() % ticks == 0 {
            func()
        } 
    }
}

fn main() {
    go set_ticker(1000, ticker)
    for {}
}
```
*run `v -g run timer.v`*, got:
```
   expected ‘void (*)()’ but argument is of type ‘int’
   2074 | void main__set_ticker(int ticks, void (*func)());
                                                            ~~~~~~~^~~~~~~
```

**Comment by ghost** [comment]
Is this a feature request or a bug report? seems just a question.
you should ask for a question at https://discord.com/channels/592103645835821068/592294828432424960

**Comment by changrui** [comment]
I mean a feature request. @kahsa
